### PR TITLE
RPC: sync committee duties

### PR DESCRIPTION
**Pull request by terencechain**
# Description

Add sync committee duties retrieval from #9123

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go [hunk 1]
```diff
@@ -178,6 +184,45 @@ func (vs *Server) duties(ctx context.Context, req *ethpb.DutiesRequest) (*ethpb.
...
+			}
+			assignment.IsSyncCommittee, err = helpers.IsCurrentPeriodSyncCommittee(s, idx)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "Could not determine current epoch sync committee: %v", err)
+			}
+			if assignment.IsSyncCommittee {
+				assignValidatorToSyncSubnet(req.Epoch, syncCommPeriod, pubKey, csc, assignment.Status)
+			}
+
+			nextSlotEpoch := helpers.SlotToEpoch(s.Slot() + 1)
```
Comment (nisdas) [inline beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go:203]: `GetDuties` is only called once every epoch at slot 0, so we would never be able to check this in time. Once duties are cached, we never request them again.

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go:203]: We shouldn't be fixated by caller behavior though. If someone were to use this and request slot - 1 then they will catch a bug. I would prefer the correct implementation here

File: beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go [hunk 2]
```diff
@@ -178,6 +184,45 @@ func (vs *Server) duties(ctx context.Context, req *ethpb.DutiesRequest) (*ethpb.
 			vStatus, _ := vs.validatorStatus(ctx, s, pubKey)
 			assignment.Status = vStatus.Status
 		}
+
+		// Are the validators in current or next epoch sync committee.
+		if postAltairTransition {
+			syncCommPeriod := helpers.SyncCommitteePeriod(req.Epoch)
```
Comment (rauljordan) [inline beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go:190]: Definitely needs to be in its own function with test coverage

File: beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go [hunk 3]
```diff
@@ -217,3 +262,51 @@ func (vs *Server) AssignValidatorToSubnet(pubkey []byte, status ethpb.ValidatorS
 	totalDuration := epochDuration * time.Duration(assignedDuration)
 	cache.SubnetIDs.AddPersistentCommittee(pubkey, assignedIdxs, totalDuration*time.Second)
 }
+
+// assignValidatorToSyncSubnet checks the status and pubkey of a particular validator
+// to discern whether persistent subnets need to be registered for them.
+func assignValidatorToSyncSubnet(currEpoch types.Epoch, syncPeriod uint64, pubkey []byte,
```
Comment (rauljordan) [inline beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go:268]: This method also does a lot and should be refactored into smaller pieces

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go:268]: @nisdas can you help for this one?

**Review by rauljordan — DISMISSED** [review]

**Review by rauljordan — APPROVED** [review]

**Pull request merged**